app.go

```diff
@@ -35,16 +39,56 @@ const (
 	PayloadChunkLast = 2
 )
 
+type LedgerAppMode int
+
+const (
+	ValidatorMode LedgerAppMode = 1 + iota
+	ConsumerMode
+)
+
+
 // LedgerOasis represents a connection to the Ledger app
 type LedgerOasis struct {
 	device  ledger_go.LedgerDevice
 	version VersionInfo
 }
 
+var logger = logging.GetLogger(LogModuleName)
+
+//SetLoggerModule sets logging module
+func (ledger *LedgerOasis)SetLoggerModule(module string) {
+	logger = logging.GetLogger(module)
+}
+
+//GetModeForRole returns the app mode compatible with role
+func GetModeForRole(role signature.SignerRole) LedgerAppMode {
+	switch role {
+	case signature.SignerConsensus:
+		return ValidatorMode
+	default:
+		return ConsumerMode
+	}
+}
+
+//GetDerivationPath return the derivation path for mode
+func GetDerivationPath(mode LedgerAppMode) []uint32 {
+	switch mode {
+	case ValidatorMode:
+		return ListingDerivationPathValidator
+	case ConsumerMode:
+		return ListingDerivationPath
+	}
+
+	return []uint32{}
+}
+
+
 // Displays existing Ledger Oasis apps by address
-func ListOasisDevices(path []uint32) {
+func ListOasisDevices(mode LedgerAppMode) {
```

Comment:
This change can be a problem:
- in some cases, apps may want to try user range of derivation paths which are all valid for the specific mode. 
We should not limit this to only two options. 
I think a good option is to use the two first items in the derivation path to define the mode or reject the call.
- it breaks API compatibility with the existing Oasis CLI
-

Reply:
ok, fixed!